serviceAccount/ServiceAccount.psm1

```diff
@@ -0,0 +1,1546 @@
+        Return $false 
+    } 
+      
+     # Execute UpdateServiceSettings.sql, but not for secondary WID. Secondary SQL never gets to this function. 
+      
+    If (!(($Role -eq "SecondaryComputer") -and ($DBMode -eq "WID"))) 
+    { 
+        Start sqlcmd.exe -ArgumentList "-S $SQLHost -i $env:Temp\ADFSSQLScripts\UpdateServiceSettings.sql -o $env:Temp\ADFSSQLScripts\UpdateServiceSettings.log" -Wait -WindowStyle Hidden | Out-File $LogPath -Append 
+      
+        If (!$?) 
+        { 
+            Write-Host "`tFailed to execute UpdateServiceSettings.sql. Exiting...." -ForegroundColor "red" 
+            ($ElapsedTime.Elapsed.ToString())+" [ERROR]     Failed to execute UpdateServiceSettings.sql" | Out-File $LogPath -Append 
+            Return $false 
+        } 
+    } 
+    Return $true 
+}
+
+function Update-AdfsServiceAccountRule
+{
+    param(
+        [parameter(Mandatory=$true, Position=1)]
+        [string]$ServiceAccount,
+
+        [parameter(ValueFromPipeline=$True)]
+        [string[]]$SecondaryServers,
+
+        [parameter()]
+        [switch]$RemoveRule    
+    )
+
+
+    #Receive user confirmation
+    $message  = "Confirmation required"
+    $question = "This script will will write to the AD FS configuration database. Are you sure you want to proceed?"
+
+    $choices = New-Object Collections.ObjectModel.Collection[Management.Automation.Host.ChoiceDescription]
+    $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&Yes'))
+    $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&No'))
+
+    $decision2 = $Host.UI.PromptForChoice($message, $question, $choices, 1)
+
+    if($decision2 -eq 1)
+    {
+        Write-Host "Terminating execution of script"
+        return
+    }
+
+    #Validate provided account exists
+    $User = $ServiceAccount
+    if($ServiceAccount -match '\\')
+    {
+        $Account = $ServiceAccount.Split('\') #Input given in the format domain\user 
+        $User = $Account[1]
+    }
+    $Lookup = Get-ADUser -Filter {Name -eq $User} 
+    if($Lookup -eq $null)
+    {
+        Write-Error "The specified account $User does not exist"
+        return
+    }
+
+
+    #Create rule with new service account
+    $SID = ConvertTo-Sid($ServiceAccount)
+    $ServiceAccountRule = "@RuleName = `"Permit Service Account`"`nexists([Type == `"http://schemas.microsoft.com/ws/2008/06/identity/claims/primarysid`", Value == `"$SID`"])`n=> issue(Type = `"http://schemas.microsoft.com/authorization/claims/permit`", value = `"true`");`n`n"
+    $Properties = Get-AdfsInternalSettings
+
+    #Backup service settings prior to adding new rule
+    $BackUpPAth = ((Convert-Path .) + "\serviceSettingsData" + "-" + (get-date -f yyyy-MM-dd-hh-mm-ss) + ".xml") -replace '\s',''
+    Get-DataContractSerializedString -object $Properties | Export-Clixml $BackUpPAth
+    Write-Host ("Backup of current service settings stored at $BackUpPath")
```

Comment:
Looks like there's a typo on $BackUpPath, sometimes there is a capital A in "Path" and sometimes there's not